crates/sui-types/src/crypto.rs

```diff
@@ -605,7 +605,8 @@ where
 {
     fn from_signable_bytes(bytes: &[u8]) -> Result<Self, Error> {
         // Remove name tag before deserialization using BCS
-        let name = serde_name::trace_name::<Self>().expect("Self must be a struct or an enum");
```

Comment:
this should 100% be a panic

Reply:
This function is part of a blanket implementation pegged on a marker trait.
If anyone slaps this marker trait on something that is not `serde_name`-able (more specifically `serde_name`-traceable), I have a problem with the delay between that mistake slipping in and the much later time at which we will discover asking for signature of any such object crashes the node. This is a textbook case of remote-crashing a family of nodes.

Reply:
Ah, never mind, there's no need to remove the panic there, I should use a sealed trait in place of this `BCSSignable` 🤦 
To be continued